File: c/chapitre_2/exo_21.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/wait.h>

int main()
{
    __pid_t pidFils ;
    int status ;

    switch (pidFils = fork() )
    {
    case -1 :
            printf("Erreur de lancement du Fork") ;
            exit (-1) ;
        break;
    case 0 :
            printf("Je suis le fils\n") ;
            exit(0);
        break;
    default :
            printf("Je suis le Pére\n") ;
            wait(&status) ;
        break ;
    }
    
    return 0 ;
}

File: python/chapitre_2/exo21.py
import os
import sys

def main():
    pidFils = os.fork()
    status = 0

    if pidFils == -1:
        print("Erreur de lancement du Fork")
        sys.exit(-1)
    elif pidFils == 0:
        print("Je suis le fils")
        os._exit(0)
    else:
        print("Je suis le Pére")
        os.wait()

if __name__ == "__main__":
    main()